toqito/state_props/is_separable.py

```diff
@@ -148,9 +155,27 @@ def is_separable(state: np.ndarray, dim: None | int | list[int] = None, level: i
     # Check these tests.
     if min_dim == 2:
         # Check if X is separable from spectrum.
-        if (lam[0] - lam[2 * max_dim - 1]) ** 2 <= 4 * lam[2 * max_dim - 2] * lam[2 * max_dim] + tol**2:
-            print("Determined to be separable by inspecting its eigenvalues.")
-            print("N. Johnston. Separability from spectrum for qubit-qudit states. Phys. Rev. A, 88:062330, 2013.")
+        if (lam[0] - lam[2 * max_dim - 2]) ** 2 <= 4 * lam[2 * max_dim - 3] * lam[2 * max_dim - 1] + tol**2:
+            return True
+
+        # For the rest of the block matrix tests, we need the 2-dimensional
+        # subsystem to be the *first* subsystem, so swap accordingly.
+        state_t = swap(state, [1, 2], dim) if dim[0] > 2 else state
+
+        # Check if Lemma 1 of "N. Johnston. Separability from spectrum for qubit-qudit states. Phys. Rev. A, 88:062330, 2013"
+        # applies to X. Also check the Hildebrand 2xn results.
+        A = state_t[:max_dim, :max_dim]
+        B = state_t[:max_dim, max_dim : 2 * max_dim]
+        C = state_t[max_dim : 2 * max_dim, max_dim : 2 * max_dim]
+
+        if np.linalg.matrix_rank(B - B.conj().T) <= 1 and is_ppt_state:
+            return True
+
+        X_2n_ppt_check = np.vstack((np.hstack(((5 / 6) * A - C / 6, B)), np.hstack((B.conj().T, (5 / 6) * C - A / 6))))
+        if is_positive_semidefinite(X_2n_ppt_check) and is_ppt(X_2n_ppt_check, 2, [2, max_dim]):
+            return True
+
+        if np.linalg.norm(B) ** 2 <= np.min(np.real(np.linalg.eig(A))) * np.min(np.real(np.linalg.eig(C))) + tol**2:
             return True
```

Comment:
Should we add a reference here in the comment to indicate where the separability condition comes from?

```py
# Determined to be separable by using Lemma 1 of reference:
# N. Johnston. Separability from spectrum for qubit-qudit states. Phys. Rev. A, 88:062330, 2013.
```